src/server/worker/datum/datum.go

```diff
@@ -161,6 +162,11 @@ func (s *Set) WithDatum(meta *Meta, cb func(*Datum) error, opts ...Option) error
 				if retErr == nil || i == d.numRetries {
 					retErr = d.finish(retErr)
 				}
+				duration := time.Duration(int64(d.meta.Stats.ProcessTime.GetNanos()) + d.meta.Stats.ProcessTime.GetSeconds()*1000000000)
```

Comment:
maybe use `time.Second` to avoid the zeros
```suggestion
				duration := time.Duration(d.meta.Stats.ProcessTime.GetNanos()) + time.Duration(d.meta.Stats.ProcessTime.GetSeconds())*time.Second
```

Reply:
good point, thanks!